src/product_config_utils.rs

```diff
@@ -181,16 +201,15 @@ where
 /// Given a `config` object and the `property_kind` vector, it uses the `Configuration::compute_*` methods
 /// to partition the configuration properties by `PropertyNameKind`.
 ///
-/// The output is map where the configuration properties are keyed by `PropertyNameKind`.
+/// The output is a map where the configuration properties are keyed by `PropertyNameKind`.
 ///
 /// # Arguments
 /// - `resource`       - Not used directly. It's passed on to the `Configuration::compute_*` calls.
 /// - `name`           - Not used directly but passed on to the `Configuration::compute_*` calls. It usually
 ///                      contains a role name or a role group name.
 /// - `config`         - The configuration properties to partition.
 /// - `property_kinds` - The "buckets" used to partition the configuration properties.
-///
-fn partition_properties_by_kind<T>(
+fn transform_properties_to_kind<T>(
```

Comment:
I'm sorry about not being more specific last time. We don't transform properties to kind here so this new name is also wrong :) 

Either keep the old name or chose something that matches what it actually does like `parse_role_config` or something along those lines or `transform_role_to_config` I don't know